process/track/baseBlockTrack.go

```diff
@@ -275,6 +276,64 @@ func (bbt *baseBlockTrack) GetCrossNotarizedHeader(shardID uint32, offset uint64
 	return bbt.crossNotarizer.getNotarizedHeader(shardID, offset)
 }
 
+// CheckBlockBasicValidity checks if the given header is valid related to some basic checks
+func (bbt *baseBlockTrack) CheckBlockBasicValidity(headerHandler data.HeaderHandler) error {
+	if check.IfNil(headerHandler) {
+		return fmt.Errorf("nil header handler")
+	}
+
+	finalHeader, _, err := bbt.GetFinalHeader(headerHandler.GetShardID())
```

Comment:
this error is misleading, this function should only check if the block is valid or invalid.
This error is unrelated to this specific block, so it shouldn't be here.

Reply:
done, split the function in 2